rasa/core/brokers/kafka.py

```diff
@@ -211,10 +209,22 @@ def _publish(self, event: Dict[Text, Any]) -> None:
 
         self.producer.send(self.topic, value=event, key=partition_key, headers=headers)
 
-    def _close(self) -> None:
-        self.producer.close()
+    def _get_partition_key(self, partition_seed: Text) -> bytes:
```

Comment:
should this be `Optional[Text]` since `event.get("sender_id")` could return `None`?

Reply:
I didn't realize "sender_id"/"conversation_id" could be a missing field in event.  I'll make the appropriate change, but just want to confirm that the missing field is a distinct possibility.

Reply:
I actually think we can rely on it being there given this line https://github.com/RasaHQ/rasa/blob/main/rasa/core/exporter.py#L254